samza-core/src/main/scala/org/apache/samza/config/loaders/PropertiesConfigLoader.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.config.loaders
+
+import java.io.FileInputStream
+import java.util.Properties
+
+import scala.collection.JavaConverters._
+import org.apache.samza.SamzaException
+import org.apache.samza.config.{Config, ConfigLoader, MapConfig}
+import org.apache.samza.util.Logging
+
+/**
+ * Load config from a properties file with given path. "path" will be given in the input properties map.
+ */
+class PropertiesConfigLoader extends ConfigLoader with Logging {
```

Comment:
Since this is a new class, can you write this in Java instead of Scala?

Reply:
Moved to Java.